types/types.d.ts

```diff
@@ -38,9 +38,13 @@ interface SessionData extends ExpressSessionData {
 
   /**
    * Regenerates the session by generating a new `sessionId`.
+   *
+   * ignoreFields specifies which fields should be kept in the new session object.
    */
   regenerate(callback: Callback): void;
+  regenerate(callback: Callback, ignoreFields: string[]): void;
```

Comment:
Isn't the callback as the first parameter odd?

Reply:
yes, but it will not introduce a breaking chance